### DISCUSSION TITLE
Default vs New York?

### DISCUSSION THREAD

**Discussion by nick-kang** [shadcn-ui/ui#2930] [Q&A]
Is there guidance on the difference between Default vs New York? 

Tried searching the docs and old commits, but not finding much.

From toggling between the two themes, it just looks like New York is a little more compact?

**Comment by pablotanner** [comment]
Yeah the difference is basically that New York uses less padding and sometimes uses less rounding/border-radius on the components.

**Comment by zzz6519003** [comment]
Good Call!

**Comment by nguyenhieptech** [comment]
You can go to **Themes** tab, click Customize / the info icon on the right of "Style" to see it.

"What is the difference between the New York and Default style?
A style comes with its own set of components, animations, icons and more.
- The Default style has larger inputs, uses lucide-react for icons and tailwindcss-animate for animations.
- The New York style ships with smaller buttons and cards with shadows. It uses icons from Radix Icons."

![image](https://github.com/shadcn-ui/ui/assets/48057064/5c6d22a1-d3a3-4532-ab30-657ce46e2e39)
![image](https://github.com/shadcn-ui/ui/assets/48057064/32780bee-7a0a-403e-ba58-461465b98895)

**Reply by mearleycf** [reply]
Not anymore you can't. As of Nov 19, 2024.

**Comment by scottrepreneur** [comment]
This is not documented afaict, but is asked on install.

<img width="545" alt="Screenshot 2024-06-25 at 11 18 21" src="https://github.com/shadcn-ui/ui/assets/1778380/8243b3ea-13eb-4eae-b4b5-f8acf0ae49e5">

**Reply by dagilleland** [reply]
And for those new to shadcn-ui, it can stop them in their tracks. If it can be changed easily later on, then it would be nice to know and people won't feel like they are stumbling out of the gate just because they don't know what they are getting/giving up in choosing between those two styles.

**Reply by dagilleland** [reply]
Ah, just found some documentation on it [here](https://www.shadcn-svelte.com/docs/components-json#style). Looks like it is a "once you choose, you cannot undo" kind of choice. 😢

**Reply by mearleycf** [reply]
That is ~~barely~~ documentation. On top of that, because they're not compared side by side, I can't tell the difference anyway. If you're going to have an option in your frigging project init CLI, you'd better explain it REALLY well in the documentation.

**Reply by Viibrant** [reply]
> And for those new to shadcn-ui, it can stop them in their tracks. If it can be changed easily later on, then it would be nice to know and people won't feel like they are stumbling out of the gate just because they don't know what they are getting/giving up in choosing between those two styles.

Precisely what is happening to me now!

**Reply by benjang032** [reply]
> > And for those new to shadcn-ui, it can stop them in their tracks. If it can be changed easily later on, then it would be nice to know and people won't feel like they are stumbling out of the gate just because they don't know what they are getting/giving up in choosing between those two styles.
> 
> Precisely what is happening to me now!

seconding that right now haha

**Comment by mintthemoon** [comment]
Yep add my name to the list of people who ended up here from a very confused search after trying to init my first project. I even thought to check the Themes tab there but never saw it still as I didn't hit customize. Should absolutely be in the docs imo, and ideally described better by the CLI in the prompt.

**Comment by pepijnolivier** [comment]
+1

I guess buildings in new york are cramped for space, so that's why they named it as such. But yes, this CLI choice could definitely benefit from a link to a page where it is explained.

To add on what @nguyenhieptech mentioned: 

- you can test the difference by navigating to https://ui.shadcn.com/themes and clicking on "customize"
- you can also test by navigating to a component and changing the style in the preview, eg: https://ui.shadcn.com/docs/components/table

**Reply by mearleycf** [reply]
I don't actually think your first bullet point works anymore--at least, when I went there today, Nov 19 2024, it did not show an option to toggle between new york and default in the customize modal.

**Comment by codemile** [comment]
Default = "you're building a landing page, website or blog"

New York = "you're building a SaaS, dashboard or have a lot of forms"

**Reply by ismaelnt** [reply]
It makes sense.

**Reply by Floriferous** [reply]
This tip could be added to the CLI, I had to go out of my way to research this choice instead of being able to build right away.

**Reply by codemile** [reply]
Renaming "Default" to "Website" and renaming "New York" to "SaaS" wouldn't require people to read the docs. The options would be self explaining.

**Reply by biapar** [reply]
How to change after from Default to NY?

**Reply by wengkitttt** [reply]
> How to change after from Default to NY?

re-run `npx shadcn@latest init`

**Reply by biapar** [reply]
Thanks. And without cli?

**Reply by biapar** [reply]
> > How to change after from Default to NY?
> 
> re-run `npx shadcn@latest init`

Is't dangerous?

**Reply by joshtenorio** [reply]
> > > How to change after from Default to NY?
> > 
> > 
> > re-run `npx shadcn@latest init`
> 
> Is't dangerous?

Maybe (I don't actually know), but assuming you're working inside of a git repository you can always just commit your work, re-run `npx shadcn@latest init`, if it works out fine you're good, and otherwise you can `git restore .` to reset your repository to before re-running the init command

**Comment by sebastiancarlos** [comment]
There is more documentation about styles on the [changelog](https://ui.shadcn.com/docs/changelog) page, under the "June 2023" update. Unfortunately the documentation search is not too good, so this is hard to find.

**Reply by bryanprimus** [reply]
nice

**Reply by mearleycf** [reply]
Oh, good. I mean, I always read 2 years worth of changelog updates to figure out how to use the installation CLI.

**Comment by musjj** [comment]
Not an official resource, but there's this nice article discussing the differences: https://www.shadcndesign.com/blog/difference-between-default-and-new-york-style-in-shadcn-ui

**Reply by mearleycf** [reply]
Again--not to keep saying the same thing, but this is not the first, or even 5th place I would look to figure out what the difference is between these two options that are presented in the FIRST STEP OF THE CLI.

**Comment by deep100x** [comment]
![image](https://github.com/user-attachments/assets/5a1f7a78-c1f3-47c4-81bb-6e9a0a6d52cd)
Default is like Minimalist, neutral design. Great for projects needing a simple, adaptable style without heavy customization. Easier to style from scratch.

On the other hand New York Offers a more refined, modern look with enhanced typography and styling. Ideal for applications that want a polished, high-end aesthetic out-of-the-box.

Both are customizable, but Default is simpler, while New York has more visual flair by default.

**Reply by LikeDreamwalker** [reply]
Don't know why but at 2024/11/11, there is no selection for Default or New York
![image](https://github.com/user-attachments/assets/82039bc8-0b2d-4687-8820-4453caa24302)

**Reply by zzz6519003** [reply]
Ghost trick

**Reply by bryanprimus** [reply]
looks like they removed it

**Reply by telio-s** [reply]
I just read this post today and want to try by myself, turn out it gone

**Reply by Zanga1** [reply]
> Don't know why but at 2024/11/11, there is no selection for Default or New York

Svelte-shadcn still has it. [svelte-shadcn/themes](https://www.shadcn-svelte.com/themes). As far as I can tell they're identical.

**Comment by aqiongbei** [comment]
So, how do I choose?

**Reply by funwarioisii** [reply]
I followed this opinion.
https://github.com/shadcn-ui/ui/discussions/2930#discussioncomment-10437514

**Comment by LikeDreamwalker** [comment]
By my understanding, if you like the style that OpenAI, Next.js and Vercel are using, which looks like mainly dark or light colors with simple outlined style, you should use New York to make your app have more designed details in it.

Otherwise if you just want to use shadcn/ui to help you build your app, or you have already got the design of your app, you should use Default.

And there are almost no difficulties in overwriting New York style, so I say using New York won't be a bad choice.

**Comment by biapar** [comment]
I suggest to write a document about how to pass from the default style to New York style after the installation.

**Comment by mehrdadrafiee** [comment]
change `style` in `components.json` to New York. Like so:

```js
{
  ...
  "style": "new-york",
  ...
}
```

**Comment by daveycodez** [comment]
The biggest difference is the focus outline. I wish we could get the New York style focus outline on the larger Default Inputs. :(

**Comment by jarodtaylor** [comment]
I'm just bummed that they're [getting rid of Default and only doing New York](https://ui.shadcn.com/docs/tailwind-v4#:~:text=We%27re%20deprecating%20the%20default%20style.%20New%20projects%20will%20use%20new%2Dyork) with the next version (`shadcn@canary`). I like the bigger elements with more padding/space. 😔

**Reply by daveycodez** [reply]
I agree, but that's the direction it seems. There's always [HeroUI](https://www.heroui.com)

**Reply by jarodtaylor** [reply]
I prefer ShadCN to HeroUI or some of the other alternatives. I'm just hoping I can easily configure my shadcn components to provide more spacing, etc. Otherwise, I'll just have to modify all of the components manually.
